MoltenVK/MoltenVK/GPUObjects/MVKPipeline.mm

```diff
@@ -1208,7 +1236,16 @@ static VkFormat mvkFormatFromOutput(const SPIRVShaderOutput& output) {
 			}
 
 			auto vaDesc = inputDesc.attributes[pVKVA->location];
-			vaDesc.format = (decltype(vaDesc.format))getPixelFormats()->getMTLVertexFormat(pVKVA->format);
+			auto mtlFormat = (decltype(vaDesc.format))getPixelFormats()->getMTLVertexFormat(pVKVA->format);
+			if (pVKVB->stride && attrSize > pVKVB->stride) {
+				/* Metal does not support overlapping loads. Truncate format vector length to prevent an assertion
+				 * and hope it's not used by the shader. */
+				auto newFormat = (decltype(vaDesc.format))mvkAdjustFormatVectorToSize((MTLVertexFormat)mtlFormat, pVKVB->stride);
+				reportError(VK_SUCCESS, "Found attribute with size (%u) larger than it's binding's stride (%u). Changing descriptor format from %u to %u.",
+					attrSize, pVKVB->stride, (uint32_t)mtlFormat, (uint32_t)newFormat);
+				mtlFormat = newFormat;
```

Comment:
To make this a little more user-friendly, instead of reporting the formats as integers, can you add to `MVKPixelFormats`:

```
const char* MVKPixelFormats::getName(MTLVertexFormat mtlFormat) {
    return getMTLVertexFormatDesc(mtlFormat).name;
}
```

and use it in the `reportError("...format from %s to %s")`?

See the similar `MVKPixelFormats` functions for the names of `VkFormat` and `MTLPixelFormat`.

Reply:
Good point, added.